source/core/src/main/com/csse3200/game/entities/factories/WeaponFactory.java

```diff
@@ -25,150 +13,102 @@
+        WeaponConfig.WeaponData weaponData = WeaponConfig.getWeaponData(specification);
+        if (weaponData == null) {
+            throw new IllegalArgumentException("Invalid ranged weapon specification: " + specification);
+        }
+
+        RangedWeapon rangedWeapon = new RangedWeapon() {
+            {
+                setDamage(weaponData.getDamage());
+                setRange(weaponData.getRange());
+                setFireRate(weaponData.getFireRate());
+                setAmmo(weaponData.getAmmo());
+                setMaxAmmo(weaponData.getMaxAmmo());
+                setReloadTime(weaponData.getReloadTime());
+            }
+
+            @Override
+            public String getName() {
+                return specification;
+            }
+
+            @Override
+            public Texture getIcon() {
+                return new Texture(weaponData.getIconPath());  // Fetch icon from config
+            }
+
+            @Override
+            public String getRangedSpecification() {
+                return specification;
+            }
+
+            @Override
+            public void shoot(Vector2 direction) {
+                if (getAmmo() > 0) {
+                    System.out.println(specification + " fired in direction: " + direction);
+                    setAmmo(getAmmo() - 1);
+                } else {
+                    System.out.println("Out of ammo! Need to reload.");
+                }
+            }
         };
+        return rangedWeapon;
     }
 
-    /**
-     * Create a new collectible weapon from a specification.
-     *
-     * @param type          the type of the weapon (melee or ranged)
-     * @param specification the specification of the weapon.
-     * @return The newly constructed collectible.
-     * @throws IllegalArgumentException if the weapon type is invalid.
-     */
     public Collectible create(Collectible.Type type, String specification) throws IllegalArgumentException {
         return switch (type) {
             case MELEE_WEAPON -> createMelee(specification);
             case RANGED_WEAPON -> createRanged(specification);
-            default -> throw new IllegalArgumentException("invalid weapon type: " + type);
+            default -> throw new IllegalArgumentException("Invalid weapon type: " + type);
         };
     }
 
-
-    /**
-     * Convert a collectible item into a collectible entity.
-     *
-     * @param collectible the item to convert
-     * @return the final entity containing the collectible.
-     * @throws IllegalArgumentException if the collectible is invalid.
-     */
     public Entity createWeaponEntity(Collectible collectible) throws IllegalArgumentException {
         try {
             if (collectible.getType() == Collectible.Type.MELEE_WEAPON) {
-                return createMeleeEntity((MeleeWeapon) collectible);
+                return WeaponConfig.createMeleeEntity((MeleeWeapon) collectible);
```

Comment:
As I have said earlier, ALL weapon creation must be done within the WeaponFactory, hence the name.